test/integration/suite_test.go

```diff
@@ -72,6 +74,9 @@ func generateKongBuilder() (*kong.Builder, []string) {
 }
 
 func TestMain(m *testing.M) {
+	var code int
+	defer os.Exit(code)
```

Comment:
Ouch, well this wouldn't work as I think you intended it to.

```go
package main

import "os"

func main() {
	code := 2
	defer os.Exit(code)

	code = 128
}
```

This will always return status code 2

Reply:
Thanks, didn't realize defer bound arguments that way.